InfrastructureBackupValidator/BackupValidationTool.psm1

```diff
@@ -0,0 +1,591 @@
+    }
+}
+
+<#
+ .Synopsis
+  List the ARM resources extracted from the backup.
+
+ .Description
+  - Get the backup chain according to the BackupID
+  - Decrypt the required snapshots to local
+  - Restore the backup into SQL server
+  - Retrieve ARM resources
+
+ .Example
+  $resources = Validate-AszBackup -BackupStorePath $backupStorePath -BackupStoreCredential $backupStoreCredential -BackupID $backupID -DecryptionCertPath $decryptionCertPath -DecryptionCertPassword $decryptionCertPassword -TempFolder $tempFolder
+  $resources = Validate-AszBackup -BackupStorePath $backupStorePath -BackupStoreCredential $backupStoreCredential -BackupID $backupID -DecryptionCertPath $decryptionCertPath -DecryptionCertPassword $decryptionCertPassword -SQLServerInstanceName $sqlServerInstanceName -SQLCredential $sqlCredential
+#>
+function Validate-AszBackup
+{
+    [CmdletBinding(SupportsShouldProcess, ConfirmImpact = 'Medium')]
+    [OutputType([System.Collections.Hashtable])]
+    param(
+        [Parameter(Mandatory = $true)]
+        [ValidateScript({$_ | Test-Path -IsValid})]
+        [String]
+        $BackupStorePath,
+
+        [Parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [PSCredential]
+        $BackupStoreCredential,
+
+        [Parameter(Mandatory = $true)]
+        [ValidateScript({[System.Guid]::TryParse($_, $([System.Management.Automation.PSReference][System.Guid]::Empty))})]
+        [String]
+        $BackupID,
+
+        [Parameter(Mandatory = $true)]
+        [ValidateScript({$_ | Test-Path -PathType Leaf})]
+        [String]
+        $DecryptionCertPath,
+        
+        [Parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [SecureString]
+        $DecryptionCertPassword,
+
+        [Parameter(Mandatory = $false)]
+        [ValidateNotNullOrEmpty()]
+        [String]
+        $SQLServerInstanceName = "localhost\SQLEXPRESS", # Use default SQL Express server instance
+
+        [Parameter(Mandatory = $false)]
+        [ValidateNotNullOrEmpty()]
+        [PSCredential]
+        $SQLCredential,
+
+        [Parameter(Mandatory = $false)]
+        [ValidateScript({$_ | Test-Path -PathType Container})]
+        [String]
+        $TempFolder = $env:TEMP
+    )
+
+    $ErrorActionPreference = "Stop"
+
+    # STEP 1: Connect to the backup store with provided credential
+    Write-Verbose $Strings.ProgressConnectBackupStore -Verbose
+    try
+    {
+        $null = New-PSDrive -Name $BackupStoreDriveName -Root $BackupStorePath -Credential $BackupStoreCredential -PSProvider FileSystem | Out-Null
+    }
+    catch
+    {
+        throw ($Strings.ErrorFailToConnectBackupStore -f $BackupStorePath, $_)
+    }
+
+    Install-Module -Name SqlServer
+    Import-Module SqlServer -DisableNameChecking
+    $sqlCommonParams = @{
+        ServerInstance = $SQLServerInstanceName
+    }
+
+    if (!$SQLCredential)
+    {
+        # STEP 2: Connect to the SQL server with Windows Authentication
```

Comment:
Is installing SQL server on the executing machine is a prerequisite?

Reply:
Yes, confirmed with PM on this. Let me add this prerequisite in README.